src/basics.md

```diff
@@ -326,13 +327,185 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-run-piped-external-commands]: #ex-run-piped-external-commands
+<a name="ex-run-piped-external-commands"></a>
+## Run piped external commands
+
+[![clap-badge]][clap] [![cat-os-badge]][cat-os]
+
+Shows the `<N>`th biggest files/subdirectories in the specified
+`<directory>`. It spawns three Unix processes. It is equivalant to
+run: `du -ah <directory> | sort -hr | head -n <N>`.
+
+```rust,no_run
+# extern crate clap;
+
+# use clap::{App, Arg, ArgMatches};
+use std::error::Error;
+use std::fs::metadata;
+use std::process::{exit, Command, Stdio};
+
+static DU: &'static str = "du";
+static DU_ARGUMENTS: &'static str = "-ah";
+
+static SORT: &'static str = "sort";
+static SORT_ARGUMENTS: &'static str = "-hr";
+
+static HEAD: &'static str = "head";
+static HEAD_ARGUMENTS: &'static str = "-n";
+
+fn main() {
+    let arguments = declare_command_line_arguments();
+    let results = parse_number_of_results(&arguments);
+    let directory = parse_directory(&arguments);
+
+    let du_process = match Command::new(DU)
+        .arg(DU_ARGUMENTS)
+        .arg(directory)
+        .stdin(Stdio::null())
+        .stdout(Stdio::piped())
+        .spawn()
+    {
+        Err(why) => {
+            println!("couldn't spawn {}: {}", DU, why.description());
+            exit(1)
+        }
+        Ok(process) => process,
+    };
+
+    let du_stdout = du_process.stdout.unwrap_or_else(|| {
```

Comment:
I would suggest using 
```rust
stdout.ok_or_else(|| "Could not capture `du` output!")?
```
and we do not need a separate variable binding. please see my comment on `head_output` below